src/main/java/uk/gov/hmcts/cmc/claimstore/services/OffersService.java

```diff
@@ -41,26 +43,36 @@ public void makeOffer(Claim claim, Offer offer, MadeBy party) {
     }
 
     public void accept(Claim claim, MadeBy party) {
+        assertMediationHasStarted(claim);
         assertSettlementIsNotReached(claim);
 
-        Settlement settlement = claim.getSettlement().orElse(new Settlement());
+        Settlement settlement = claim.getSettlement().get();
         settlement.accept(party);
+
         offersRepository.acceptOffer(claim.getId(), jsonMapper.toJson(settlement), LocalDateTime.now());
         eventProducer.createOfferAcceptedEvent(claim, party);
     }
 
     public void reject(Claim claim, MadeBy party) {
+        assertMediationHasStarted(claim);
         assertSettlementIsNotReached(claim);
 
-        Settlement settlement = claim.getSettlement().orElse(new Settlement());
+        Settlement settlement = claim.getSettlement().get();
```

Comment:
`.get` shouldn't be called without having a check right before it or doing an `orElseThrow`

Reply:
The check is done in the assertion. I suppose we have no option but to satisfy Sonar and do `.orElseThrow(new ConflictException("Offer has not been made yet."))`.

Reply:
its not just a sonar thing, `.get` on its own is dodgy, even if you are checking it above. 
It's not obvious

Reply:
ok